### ISSUE TITLE
Unable to replicate VGG's very deep results

### ISSUE THREAD

**Issue by olivierhenaff** [szagoruyko/loadcaffe#34]
Hi Sergei, 

I've downloaded the 16-layer VGG network and wanted to replicate their results, but after running it on the ILSVRC-2014 validation set, I get very low accuracy which makes me think I am not using it correctly. 

I understand that I need to rescale images to the [0, 255] range, permute the color channels from RGB to BGR, then subtract the mean pixel. Nevertheless, I still get very bad results. 

Have you replicated their classification results, and if so, could you share what sort of preprocessing you used? 

Thank you very much in advance,

Olivier

**Comment by szagoruyko** [comment]
Hi Olivier, I never evaluated loadcaffe models on the full ILSVRC val set. Check how it's done here https://github.com/torch/tutorials/blob/master/7_imagenet_classification/classify.lua, also that your synset mapping from 1000 outputs to actual classes is the same.

**Comment by olivierhenaff** [comment]
Sure enough, the problem was in the indexing of the synsets. The caffe models were using the dictionary in the tutorial that you graciously provided, which is different from the dictionary used in the ILSVRC development kit. 

In any case, thanks a lot for your help!

Olivier

**Comment by andrefaraujo** [comment]
Hi Sergey: I would like to suggest that you write a note on the README of this repository to make clear that the models, after loaded, must have their RGB convolution weights permuted. This comes from the models being downloaded from Caffe's zoo, which uses the BGR convention.

It would be helpful for Torch beginners like me who are not familiar with Caffe :) -- I just spent quite some time on figuring out this issue... I can send you a code snippet if you wish

**Comment by andrefaraujo** [comment]
For future users with the same question: the synset numbers and ground-truth for validation/train sets, as used for the models available in the caffe zoo, can be downloaded as done by the script found here: https://github.com/BVLC/caffe/blob/master/data/ilsvrc12/get_ilsvrc_aux.sh . The pre-trained models from the Caffe zoo will NOT predict according to the synset numbering found on the ImageNet website!

**Comment by szagoruyko** [comment]
@andrefaraujo take a look at https://github.com/szagoruyko/imagenet-validation.torch, it has the code to reproduce caffe zoo numbers

**Comment by andrefaraujo** [comment]
@szagoruyko that looks great! -- I had completely missed it, thx for the pointer. If only this link were in the loadcaffe's README, or in the torch's cheatsheet...

**Comment by szagoruyko** [comment]
@andrefaraujo fair enough, I added the link to readme, thanks
